contrib/mariner/compile.patch

```diff
@@ -0,0 +1,44 @@
+Submodule tpm/aziot-tpm-sys/azure-iot-hsm-c/deps/c-shared contains modified content
```

Comment:
Call it something more descriptive please. `gcc-11.patch` perhaps.

Reply:
changed name to be gcc-11.patch